### PR TITLE
Add support for getLocales in Polyglot i18nProvider

### PR DESCRIPTION
**Pull request by fzaninotto**
Follows #7758

### CODE REVIEW

File: docs/TranslationSetup.md [hunk 1]
```diff
@@ -54,46 +54,28 @@ import fr from 'ra-language-french';
...
+    'en', // default locale
+    [
+        { locale: 'en', name: 'English' },
+        { locale: 'fr', name: 'Français' }
+    ],
 );
 ```
 
-Then, pass the custom App Bar to a custom `<Layout>`, and the `<Layout>` to your `<Admin>`:
+The second argument to the `polyglotI18nProvider` function is the default locale. The third is the list of supported locales - and is used by the `<LocaleMenuButton>` component to display a list of languages.
```
Comment (djhi) [inline docs/TranslationSetup.md:67]:
```suggestion
The second argument to the `polyglotI18nProvider` function is the default locale. The third is the list of supported locales - and is used by the [`<LocaleMenuButton>`](./LocalesMenuButton.md) component to display a list of languages.
```

**Review by djhi — CHANGES_REQUESTED** [review]
:+1:

File: examples/simple/src/i18nProvider.tsx [hunk 2]
```diff
@@ -1,15 +1,23 @@
 import polyglotI18nProvider from 'ra-i18n-polyglot';
 import englishMessages from './i18n/en';
+import frenchMessages from './i18n/fr';
```
Comment (WiXSL) [inline examples/simple/src/i18nProvider.tsx:3]: Lint warning

File: packages/ra-i18n-polyglot/src/index.ts [hunk 3]
```diff
@@ -19,11 +19,16 @@ type GetMessages = (
...
+ *     [{ locale: 'en', name: 'English' }, { locale: 'fr', name: 'Français' }]
+ * )
  */
 export default (
     getMessages: GetMessages,
     initialLocale: string = 'en',
+    availableLocales: Locale[] = [{ locale: 'en', name: 'English' }],
     polyglotOptions: any = {}
 ): I18nProvider => {
     let locale = initialLocale;
```
Comment (WiXSL) [inline packages/ra-i18n-polyglot/src/index.ts:34]: Shouldn't we throw a warning if the `initialLocal` is not in the `availableLocales`? Maybe I'm nitpicking here

**Review by WiXSL — CHANGES_REQUESTED** [review]

**Review by WiXSL — CHANGES_REQUESTED** [review]
`TranslationSetup.md` - line 152:
The specified anchor `using-specific-polyglot-features` is in the README of polyglot package, not above in the document.

This solution is all-or-nothing: you can't silence only *some* missing translation warnings. An alternative solution consists of passing a default translation using the `_` translation option, as explained in the **Using Specific Polyglot Features section(#using-specific-polyglot-features)** above.

**Review by djhi — APPROVED** [review]

**Review by slax57 — APPROVED** [review]

**Review by WiXSL — APPROVED** [review]

**Pull request merged**